hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/NearCachedClientCacheProxy.java

```diff
@@ -833,35 +823,37 @@ public void handle(ClientMessage clientMessage) {
             extends CacheAddNearCacheInvalidationListenerCodec.AbstractEventHandler
             implements EventHandler<ClientMessage> {
 
-        private final RepairingHandler repairingHandler;
-
-        public RepairableNearCacheEventHandler() {
-            getRepairingTask().deregisterHandler(nameWithPrefix);
-            this.repairingHandler = getRepairingTask().registerAndGetHandler(nameWithPrefix, nearCache);
-        }
+        private volatile RepairingHandler repairingHandler;
 
         @Override
         public void beforeListenerRegister() {
-
+            if (supportsRepairableNearCache()) {
+                RepairingTask repairingTask = getContext().getRepairingTask(CacheService.SERVICE_NAME);
+                repairingHandler = repairingTask.registerAndGetHandler(nameWithPrefix, nearCache);
+            } else {
+                RepairingTask repairingTask = getContext().getRepairingTask(MapService.SERVICE_NAME);
```

Comment:
`MapService` in a cache proxy looks wrong. Should this both just be `CacheService.SERVICE_NAME`?

Reply:
very good finding. fixed